httpcore/_async/http2.py

```diff
@@ -210,6 +210,16 @@ async def receive_events(self, timeout: TimeoutDict) -> None:
         Read some data from the network, and update the H2 state.
         """
         data = await self.socket.read(self.READ_NUM_BYTES, timeout)
+
+        if data == b"":
+            # The server has disconnected. `h11` deals with this "empty data" signal
+            # correctly, but h2 doesn't. So we raise an exception equivalent to
+            # what would be raised in the HTTP/1.1 case.
+            raise RemoteProtocolError(
+                "peer closed connection without sending complete message body "
+                "(incomplete chunked read)"
+            )
```

Comment:
In HTTP/2 this case isn't to do with chunked data, or incomplete message bodies.
It's strictly a hard connection close, terminating all streams.
So, I reckon we can go with simpler wording here?...

```suggestion
            raise RemoteProtocolError("Connection was closed by the server.")
```